dotnet/src/SemanticKernel.Core/Functions/InstrumentedSKFunction.cs

```diff
@@ -18,153 +18,92 @@
-    /// Instance of <see cref="Counter{T}"/> to keep track of the number of successful function executions.
-    /// </summary>
-    private readonly Counter<int> _executionSuccessCounter;
-
-    /// <summary>
-    /// Instance of <see cref="Counter{T}"/> to keep track of the number of failed function executions.
-    /// </summary>
-    private readonly Counter<int> _executionFailureCounter;
-
-    /// <summary>
-    /// Wrapper for instrumentation to be used in multiple invocation places.
-    /// </summary>
-    /// <param name="func">Delegate to instrument.</param>
-    private async Task<FunctionResult> InvokeWithInstrumentationAsync(Func<Task<FunctionResult>> func)
-    {
-        using var activity = s_activitySource.StartActivity($"{this.PluginName}.{this.Name}");
-
-        this._logger.LogInformation("{PluginName}.{FunctionName}: Function execution started.", this.PluginName, this.Name);
-
-        var stopwatch = new Stopwatch();
-        stopwatch.Start();
-
-        FunctionResult result;
+        this._logger.LogInformation("Function invoking");
+        TagList tags = new() { { "function.name", this.Name } };
 
+        long startingTimestamp = Stopwatch.GetTimestamp();
         try
         {
-            result = await func().ConfigureAwait(false);
+            var result = await this._function.InvokeAsync(context, requestSettings, cancellationToken).ConfigureAwait(false);
+
+            if (this._logger.IsEnabled(LogLevel.Trace))
+            {
+                this._logger.LogTrace("Function succeeded: {Result}", result.GetValue<object>()); // Sensitive data, logging as trace, disabled by default
+            }
+            else if (this._logger.IsEnabled(LogLevel.Information))
+            {
+                this._logger.LogInformation("Function succeeded.");
+            }
+
+            return result;
         }
         catch (Exception ex)
         {
-            this._logger.LogWarning("{PluginName}.{FunctionName}: Function execution status: {Status}",
-                this.PluginName, this.Name, "Failed");
-
-            this._logger.LogError(ex, "{PluginName}.{FunctionName}: Function execution exception details: {Message}",
-                this.PluginName, this.Name, ex.Message);
-
-            this._executionFailureCounter.Add(1);
-
+            this._logger.LogError(ex, "Function invocation failed: {Message}", ex.Message);
+            tags.Add("error.type", ex.GetType().FullName);
             throw;
         }
         finally
         {
-            stopwatch.Stop();
-            this._executionTotalCounter.Add(1);
-            this._executionTimeHistogram.Record(stopwatch.ElapsedMilliseconds);
+            TimeSpan duration = new((long)((Stopwatch.GetTimestamp() - startingTimestamp) * (10_000_000.0 / Stopwatch.Frequency)));
+            this._logger.LogInformation("Function invocation duration: {Duration}ms", duration.TotalMilliseconds);
+            s_invocationDuration.Record(duration.TotalSeconds, in tags);
```

Comment:
Do we also want to add the tags to the activity?

Reply:
The tags are passed to StartActivity, right?  The only tags being used with the metrics here are the function name, and if it fails, the exception type.  The latter isn't available when StartActivity is called, and the name of the function is being used as the name of the activity, so I'm not sure there's anything else to use as a tag?

Reply:
Though addressing Mikel's feedback below, then it would no longer be the name, so the name could be used as a tag, I guess.